modules/tensor_mechanics/include/materials/ComputeAxisymmetricRZSmallStrain.h

```diff
@@ -0,0 +1,32 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef COMPUTEAXISYMMETRICRZSMALLSTRAIN_H
+#define COMPUTEAXISYMMETRICRZSMALLSTRAIN_H
+
+#include "ComputeStrainBase.h"
+
+/**
+ * ComputeAxisymmetricRZSmallStrain defines small strains in an Axisymmetric system.
+ * Within this material, '_disp_x' refers to displacement in the radial direction,
+ * u_r, and '_disp_y' refers to displacement in the axial direction, u_z.
+ * The COORD_TYPE in the Problem block must be set to RZ.
+ */
+class ComputeAxisymmetricRZSmallStrain : public ComputeStrainBase
+{
+public:
+  ComputeAxisymmetricRZSmallStrain(const std::string & name, InputParameters parameters);
+
+protected:
+  virtual void computeProperties();
+
+  VariableValue & _disp_x;
+
+  const MaterialProperty<RankTwoTensor> & _stress_free_strain;
```

Comment:
Sorry for not spotting this earlier, but the stress free strain should not be part of this material. Please add a second material that inherits form ```ComputeStressFreeStrainBase``` to provide this. The goal of the new plug and play tensor mechanics system is to make materials as modular as possible and have each material focus on a single aspect of the calculation.